### PR TITLE
layers: Reenable descriptor placement new optimizations

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**

### CODE REVIEW

**Review by Tony-LunarG — APPROVED** [review]
LGTM and gets back the rdr2 performance

**Review by ncesario-lunarg — APPROVED** [review]
Nice!

**Review by jzulauf-lunarg — COMMENTED** [review]
Clean up vestigial debug code?

File: layers/vk_layer_data.h [hunk 1]
```diff
@@ -173,6 +173,17 @@ class small_vector {
...
 
+    small_vector(size_type size, const value_type& value = value_type()) : size_(0), capacity_(N) {
+        reserve(size);
+        auto dest = GetWorkingStore();
+        for (size_type i = 0; i < size; i++) {
+            new (dest) value_type(value);
+            ++dest;
+        }
+        size_ = size;
+    }
```
[the remark this answers is not in the document]
Comment (jeremyg-lunarg) [reply layers/vk_layer_data.h:184]: DebugUpdateWorkingStore() gets called in reserve(),  which is the first line of this constructor.

**Pull request merged**